src/dwn/interfaces/records.js

```diff
@@ -0,0 +1,103 @@
+import { DwnConstant } from '@tbd54566975/dwn-sdk-js';
+
+import { Interface } from './interface.js';
+import { Record } from '../models/record.js';
+import { dataToBytes } from '../../utils.js';
+
+export class Records extends Interface {
+  constructor(dwn) {
+    super(dwn, 'Records');
+  }
+
+  get create() {
+    return this.write;
+  }
+
+  async createFrom(target, request) {
+    const { author: inheritedAuthor, target: _, ...inheritedProperties } = request.record.toJSON();
+
+    // If `data` is being updated then `dataCid` and `dataSize` must not be present.
+    if (request?.data !== undefined) {
```

Comment:
there's no need to do `request?.` as you've already assumed it's an object above by `request.record`

ditto below

Reply:
Addressed in 96c7a8030cff76db65305578bc3beb34ffc6fc51